TMGToolbox/src/assignment/preprocessing/remove_redundant_functions.py

```diff
@@ -0,0 +1,257 @@
+            self._CompareFcns(self.ExportFile, fdlist_db, ftlist_db, fplist_db, fdlist_nt, ftlist_nt, fplist_nt)
+
+            self.TRACKER.completeTask()
+
+    ##########################################################################################################    
+    
+    #----SUB FUNCTIONS---------------------------------------------------------------------------------  
+    
+    def _GetAtts(self):
+        atts = {
+                "Scenario" : str(self.Scenario.id),
+                "Export File": self.ExportFile,
+                "Version": self.version, 
+                "self": self.__MODELLER_NAMESPACE__}
+            
+        return atts 
+
+
+    # check functions in the Emme database
+    def _CheckDatabase(self):
+        fcns = _M.emmebank.functions()
+        fdlist = []
+        ftlist = []
+        fplist = []
+        folist = []
+
+        for f in fcns:
+            f_type = str(f)[0:2]
+            if f_type == "fd":
+                fdlist.append(str(f)[2:])
+            elif f_type == "ft":
+                ftlist.append(str(f)[2:])
+            elif f_type == "fp":
+                fplist.append(str(f)[2:])
+            else:
+                folist.append(str(f)[2:])
+                    
+        fdlist_database = list(map(int, fdlist))
+        ftlist_database = list(map(int, ftlist))
+        fplist_database = list(map(int, fplist))
+        folist_database = list(map(int, folist))
+
+        print "There are %s Auto Volume Delay (fd) functions in the database." %len(fdlist_database)
+        print "There are %s Transit Time (ft) functions in the database." %len(ftlist_database)
+        print "There are %s Turn Penalty (fp) functions in the database." %len(fplist_database)
+        print "There are %s other types of functions in the database." %len(folist_database)
+
+        return fdlist_database, ftlist_database, fplist_database, folist_database
+
+    # check functions used in all scenarios/networks
+    def _CheckNetworks(self):
+        scens = _M.emmebank.scenarios()
+        fdlist_net =[]
+        ftlist_net =[]
+        fplist_net =[]
+
+        for scen_id in scens:
+            _net = _M.emmebank.scenario(scen_id).get_network()
+            lks = _net.links()
+            segs = _net.transit_segments()
+            trns = _net.turns()
+                
+        for l in lks:
+            l_fcn = l.volume_delay_func 
+            if (l_fcn > 0) and (l_fcn not in fdlist_net):
+                fdlist_net.append(l_fcn)
+                        
+        for sg in segs:
+            sg_fcn = sg.transit_time_func 
+            if (sg_fcn > 0) and (sg_fcn not in ftlist_net):
+                ftlist_net.append(sg_fcn)
+                        
+        for r in trns:
+            r_fcn = r.penalty_func 
+            if (r_fcn > 0) and (r_fcn not in fplist_net):
+                fplist_net.append(r_fcn)
```

Comment:
This looks like a logical error.  Only the links/segments/etc from the last enumerated scenario will be used.